columnar/src/columnar/reader/mod.rs

```diff
@@ -102,30 +102,41 @@ impl ColumnarReader {
     pub fn num_rows(&self) -> RowId {
         self.num_rows
     }
-
-    // TODO Add unit tests
-    pub fn list_columns(&self) -> io::Result<Vec<(String, DynamicColumnHandle)>> {
+    // Iterate over the columns in a sorted way
+    pub fn iter_columns(
+        &self,
+    ) -> io::Result<impl Iterator<Item = (String, DynamicColumnHandle)> + '_> {
         let mut stream = self.column_dictionary.stream()?;
-        let mut results = Vec::new();
-        while stream.advance() {
-            let key_bytes: &[u8] = stream.key();
-            let column_code: u8 = key_bytes.last().cloned().unwrap();
-            let column_type: ColumnType = ColumnType::try_from_code(column_code)
-                .map_err(|_| io_invalid_data(format!("Unknown column code `{column_code}`")))?;
-            let range = stream.value().clone();
-            let column_name =
+        Ok(std::iter::from_fn(move || {
+            if stream.advance() {
+                let key_bytes: &[u8] = stream.key();
+                let column_code: u8 = key_bytes.last().cloned().unwrap();
+                // TODO Error Handling. The API gets quite ugly when returning the error here, so
```

Comment:
Does "quite ugly" refer to wrapping the iterator item in a `io::Result`? Is there really any alternative to doing that?

Reply:
Yes, the wrapping an the usage, since it won't be simply collected in a `Result<>`